force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,9 +238,13 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
         // Subtract the difference in years
-
+       // Integer year3 = Math.abs(year2-year1);
+        if(d1 != null && d2 != null){
+            Integer year1 = d1.year();
+            Integer year2 = d2.year();
+            return Math.abs(year2-year1);
+    }
         return null; // Replace null with the variable you used to store the result
```

Comment:
Good job. A lot of times what I like to to leave what I consider my main code path out of the conditional logic. I think this can help with readability.  I would test for the null if my if statement. 

```
  public static Integer calculateYearDifference(Date d1, Date d2) {
        if (d1 == null || d2 == null) {
            return null;
        }
        return d2.year() - d1.year();
        // Subtract the difference in years
    }
    ```